### PR TITLE
Update bitbucket related documentation with known issues and process workarounds

### PR DESCRIPTION
**Pull request by mshaposhnik**
> Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR.

### What does this PR do?
Update bitbucket related documentation with some notices about using base64 tooling and link to self-signed reposirories usage configuration.

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/RHDEVDOCS-2601

### Specify the version of the product this PR applies to.
2.27

### PR Checklist

As the author of this Pull Request I made sure that:

- [x] `vale` has been run successfully against the PR branch
- [x] Link checker has been run successfully against the PR branch
- [x] Documentation describes a scenario that is already covered by QE tests, otherwise an issue has been created and acknowledged by Che QE team
- [ ] Changed article references are updated where they are used (or a redirect has been set up on the docs side):
    - [ ] Dashboard [branding.json](https://github.com/eclipse/che-dashboard/blob/master/src/components/branding/branding.json)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)

### CODE REVIEW

**Review by skabashnyuk — APPROVED** [review]

File: modules/administration-guide/partials/proc_configuring_bitbucket_authentication.adoc [hunk 1]
```diff
@@ -54,6 +54,10 @@ The main parts of the secret are:
 
 |===
 
+NOTE: Please note that encoding string into base64 format using *base64* tool on Linux machines most probably will
+add the newline character to the end of source string, so hash may become unusable as the authentication header value after decoding.
+To avoid that, it is better to use online tools, or strip newlines explicilty (for example, using *tr -d \\n*)
```
Comment (sparkoo) [inline modules/administration-guide/partials/proc_configuring_bitbucket_authentication.adoc]: you can use `base64 -w0` and output will be without new lines

File: modules/administration-guide/partials/proc_configuring_bitbucket_authentication.adoc [hunk 2]
```diff
@@ -54,6 +54,10 @@ The main parts of the secret are:
 
 |===
 
+NOTE: Please note that encoding string into base64 format using *base64* tool on Linux machines most probably will
+add the newline character to the end of source string, so hash may become unusable as the authentication header value after decoding.
```
Comment (sparkoo) [inline modules/administration-guide/partials/proc_configuring_bitbucket_authentication.adoc]: I find "source string" confusing. Isn't it rather base64 encoded output string?

Comment (sparkoo) [inline modules/administration-guide/partials/proc_configuring_bitbucket_authentication.adoc]: Is it really "hash"?

Comment (mshaposhnik) [reply modules/administration-guide/partials/proc_configuring_bitbucket_authentication.adoc]: sometimes it is both in encoded and decoded. You can try:

`echo 'NDc2MDc3NzczMTM4Ohe2i6Ji50ygk8CYeGZPCgh03a6d' | base64 >> out.txt`
there is newline in the file. and if you take a result and decode it back:
`echo 'TkRjMk1EYzNOemN6TVRNNE9oZTJpNkppNTB5Z2s4Q1llR1pQQ2doMDNhNmQK' | base64 -d >> out.txt`
you will see newline in decoded value and that's why header didn't work. And even `-w0` during decode not helps.
(use HEX mode to view file, `0A` is a `\n`)

File: modules/administration-guide/partials/proc_configuring_bitbucket_authentication.adoc (hunk 1 above)
Comment (MichalMaler) [reply modules/administration-guide/partials/proc_configuring_bitbucket_authentication.adoc]: @sparkoo Hi Michal, can I add the `base64 -w0` as a secondary option to `tr -d \\n`? Thx

**Comment by MichalMaler** [comment]
Hi Max. Thx for the PR. Here are some suggestions. I Will add the vale errors now, but you can already take a look:
https://github.com/mshaposhnik/che-docs/pull/2

**Comment by MichalMaler** [comment]
Ok, all checks are green in my PR. Please check for tech correctness and merge at will.

**Review by MichalMaler — APPROVED** [review]

**Comment by MichalMaler** [comment]
Needed for CRW 2.6

**Pull request merged**